model/textparse/protobufparse.go

```diff
@@ -347,6 +348,24 @@ func (p *ProtobufParser) Exemplar(ex *exemplar.Exemplar) bool {
 	return true
 }
 
+func (p *ProtobufParser) CreatedTimestamp(ct *types.Timestamp) bool {
+	var foundCT *types.Timestamp
+	switch p.mf.GetType() {
+	case dto.MetricType_COUNTER:
+		foundCT = p.mf.GetMetric()[p.metricPos].GetCounter().GetCreatedTimestamp()
+	case dto.MetricType_SUMMARY:
+		foundCT = p.mf.GetMetric()[p.metricPos].GetSummary().GetCreatedTimestamp()
+	case dto.MetricType_HISTOGRAM, dto.MetricType_GAUGE_HISTOGRAM:
+		foundCT = p.mf.GetMetric()[p.metricPos].GetHistogram().GetCreatedTimestamp()
+	default:
+	}
+	if foundCT == nil {
+		return false
+	}
+	*ct = *foundCT
+	return true
```

Comment:
There is a way to write this so we don't have to define foundCT variable - perhaps it would be more readable?

```suggestion
	switch p.mf.GetType() {
	case dto.MetricType_COUNTER:
		*ct = p.mf.GetMetric()[p.metricPos].GetCounter().GetCreatedTimestamp()
	case dto.MetricType_SUMMARY:
		*ct = p.mf.GetMetric()[p.metricPos].GetSummary().GetCreatedTimestamp()
	case dto.MetricType_HISTOGRAM, dto.MetricType_GAUGE_HISTOGRAM:
		*ct  = p.mf.GetMetric()[p.metricPos].GetHistogram().GetCreatedTimestamp()
	}
	return ct != nil
```

Reply:
I tried a few combinations without success :/

`*ct = p.mf.GetMetric()[p.metricPos].GetCounter().GetCreatedTimestamp()` doesn't compile with the error:
```
cannot use p.mf.GetMetric()[p.metricPos].GetCounter().GetCreatedTimestamp() (value of type *types.Timestamp) as types.Timestamp value in assignment
```

`*ct = *p.mf.GetMetric()[p.metricPos].GetCounter().GetCreatedTimestamp()` causes a nilpointer exception if the metric doesn't have a created timestamp set.

So far, with this extra variable was the only way I found to make it work 😕

Reply:
some pointer magic needed - it is possible I think, but let's merge simplified version then.